wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -5,6 +5,9 @@
         aria-labelledby="title-{{ id }}"
         aria-hidden="true"
         class="w-dialog"
+        {% if data_url %}
+            data-url="{{ data_url }}"
+        {% endif %}
```

Comment:
I think it makes more sense to put `data-url` here instead of in the `dialog_toggle.html` (like in #8558) because the data url is used to load the dialog's content.